plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3369,4 +3368,189 @@ public void testTargetMaxFileSize()
                 // so just to be safe we check if it is not much bigger
                 .forEach(row -> assertThat((Long) row.getField(0)).isBetween(1L, maxSize.toBytes() * 3));
     }
+
+    @Test
+    public void testVacuumParameterValidation()
+    {
+        assertQueryFails(
+                "ALTER TABLE no_such_table_exists EXECUTE VACUUM",
+                "\\Qline 1:1: Table 'iceberg.tpch.no_such_table_exists' does not exist");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33')",
+                "\\QUnable to set catalog 'iceberg' table procedure 'VACUUM' property 'retention_threshold' to ['33']: duration is not a valid data duration string: 33");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33mb')",
+                "\\QUnable to set catalog 'iceberg' table procedure 'VACUUM' property 'retention_threshold' to ['33mb']: Unknown time unit: mb");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33s')",
+                "\\QRetention specified (33.00s) is shorter than the minimum retention configured in the system (7.00d). Minimum retention can be changed with iceberg.vacuum.min-retention configuration property or iceberg.vacuum_min_retention session property");
+    }
+
+    @Test
+    public void testVacuum()
+            throws Exception
+    {
+        String tableName = "test_vacuuming_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = Session.builder(getSession())
+                .setCatalogSessionProperty("iceberg", "vacuum_min_retention", "0s")
+                .build();
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer)");
```

Comment:
This is a really nit-picky test thing, but can you have a test with explicit table locations. One where the path ends in a `/` and one where it does not?

I'm worried about the path matching cleanup